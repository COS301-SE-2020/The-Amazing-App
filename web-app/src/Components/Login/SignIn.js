import React, { useState } from "react";
import { Button, Form, Grid, Message, Segment } from "semantic-ui-react";
import image from "../../Assets/logo.png";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { signup, signin, resetPassword } from "../../store/actions/auth";
import useForm from "../../utils/useForm";
import validate from "../../utils/validateLoginForm";

const Login = ({
  signup,
  signin,
  resetPassword,
  authMsg,
  history,
  loading,
}) => {
  const [newUser, setNewUser] = useState(false);
  const [reset, SetReset] = useState(false);
  const [credentials, handleChange, handleSubmit, errors] = useForm(
    login,
    validate,
    reset
  );

  function login() {
    if (newUser) {
      // signup
      signup(credentials.email, credentials.password);
    } else {
      if (reset) {
        // reset password
        resetPassword(credentials.email);
      } else {
        // signin
        signin(credentials.email, credentials.password, () =>
          history.push("/")
        );
      }
    }
  }

  return (
    <div className="login">
      <h2>
        {reset ? "Reset password" : newUser ? "Create an account" : "Sign in"}
      </h2>
      {authMsg && <p className="auth-message">{authMsg}</p>}
      <Grid
        textAlign="center"
        style={{ height: "100vh" }}
        verticalAlign="middle"
      >
        <Grid.Column style={{ maxWidth: 450 }}>
          <Form size="medium" onSubmit={handleSubmit} noValidate>
            <Segment stacked>
              <div style={{ alignItems: "center" }}>
                <img
                  src={image}
                  style={{ width: 130, height: 130, alignSelf: "center" }}
                />
              </div>
              <div>
                <Form.Input
                  fluid
                  icon="envelope"
                  type="email"
                  name="email"
                  iconPosition="left"
                  placeholder="E-mail address"
                  id="useremail"
                  value={credentials.email}
                  onChange={handleChange}
                  className={
                    (errors.emailIsEmpty || errors.emailFormatInvalid) &&
                    "input-error"
                  }
                />
                {errors.emailIsEmpty && <small>{errors.emailIsEmpty}</small>}
                {errors.emailFormatInvalid && (
                  <small>{errors.emailFormatInvalid}</small>
                )}
              </div>

              {!reset && (
                <div>
                  <Form.Input
                    fluid
                    icon="lock"
                    iconPosition="left"
                    placeholder="Password"
                    name="password"
                    value={credentials.password}
                    type="password"
                    id="userpassword"
                    onChange={handleChange}
                    className={
                      (errors.passIsStrong || errors.passIsEmpty) &&
                      "input-error"
                    }
                  />
                  {errors.passIsStrong && <small>{errors.passIsStrong}</small>}
                  {errors.passIsEmpty && <small>{errors.passIsEmpty}</small>}
                </div>
              )}

              <Button id="loginbtn" fluid color="teal" size="medium">
                Login
              </Button>
              <Message>
                Don't have an account ?<Link to="/signup"> Register</Link>
              </Message>
            </Segment>
          </Form>
        </Grid.Column>
      </Grid>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    authMsg: state.authReducer.authMsg,
    loading: state.apiStatusReducer.apiCallsInProgress > 0,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    signup: (email, password) => dispatch(signup(email, password)),
    signin: (email, password, callback) =>
      dispatch(signin(email, password, callback)),
    resetPassword: (email) => dispatch(resetPassword(email)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);

/*class SignIn extends React.Component {
  state = { email: "", pasword: "", err: "" };
  onFormSubmit = (event) => {
    event.preventDefault();

    const data = {
      email: this.state.email,
      password: this.state.password,
    };

    this.props.signIn(data);
  };

  render() {
    return (
      <Grid
        textAlign="center"
        style={{ height: "100vh" }}
        verticalAlign="middle"
      >
        <Grid.Column style={{ maxWidth: 450 }}>
          <Form size="medium" onSubmit={this.onFormSubmit}>
            <Segment stacked>
              <div style={{ alignItems: "center" }}>
                <img
                  src={image}
                  style={{ width: 130, height: 130, alignSelf: "center" }}
                />
              </div>
              <Form.Input
                fluid
                icon="envelope"
                type="email"
                iconPosition="left"
                placeholder="E-mail address"
                id="useremail"
                onChange={(e) => this.setState({ email: e.target.value })}
              />
              <Form.Input
                fluid
                icon="lock"
                iconPosition="left"
                placeholder="Password"
                type="password"
                id="userpassword"
                onChange={(e) => this.setState({ password: e.target.value })}
              />

              <Button id="loginbtn" fluid color="teal" size="medium">
                Login
              </Button>
              <Message>
                Don't have an account ?<Link to="/signup"> Register</Link>
              </Message>
            </Segment>
          </Form>
        </Grid.Column>
      </Grid>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    signIn: (creds, callback) => dispatch(signIn(creds, callback())),
  };
}

export default connect(null, mapDispatchToProps)(SignIn);*/
