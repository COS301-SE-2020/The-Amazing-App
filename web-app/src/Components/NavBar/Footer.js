import React from "react";
import { MDBContainer, MDBFooter } from "mdbreact";

const Footer = () => {
  return (
    <div className="fixed-bottom">
      <MDBFooter
      className="font-small fixed-bottom"
      style={{
        backgroundColor: "#2A9D8F",
        position: "absolute",
        width: "100%",
        bottom: 0,
        height: "3.5em",
      }}
    >
      <div className="footer-copyright text-center py-3">
        <MDBContainer fluid style={{ color: "white" }}>
          All Rights Reserved<a href="#">. The Amazing App </a>&copy;{" "}
          {new Date().getFullYear()}
        </MDBContainer>
      </div>
    </MDBFooter>
    </div>
  );
};

export default Footer;
