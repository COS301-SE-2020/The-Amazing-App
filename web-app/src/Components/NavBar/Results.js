import React from "react";
import {Header, Modal,  Form, Button } from "semantic-ui-react";
import Cookies from 'js-cookie';
import axios from 'axios';
export default axios.create(
  {
    baseURL : 'https://console.firebase.google.com/project/the-amazing-app-77843/firestore/data/Users/1jcyEtFSAflH0Re4Grxx' 
  }
)