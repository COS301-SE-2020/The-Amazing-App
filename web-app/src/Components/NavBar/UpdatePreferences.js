import React from "react";
import {Header, Modal,  Form, Button } from "semantic-ui-react";



export default class UpdatePreferences extends React.Component {

  constructor(props) {
    super(props)

    this.onChangeUserName = this.onChangeUserName.bind(this);
    this.onChangeUserEmail = this.onChangeUserEmail.bind(this);
    this.onChangeUserPassword = this.onChangeUserPassword.bind(this);
    this.onChangeConfirmPassword = this.onChangeUserPassword.bind(this);
    this.onSubmitUsername = this.onSubmitUsername.bind(this);
    this.onSubmitEmail = this.onSubmitEmail.bind(this);
    this.onSubmitPassword = this.onSubmitPassword.bind(this);
    this.state = {
      username: 'TheTshegofatso',
      email: 'Tshegofatso@gmail.com',
      password: '*******',
      newusername: '',
      newemail: '',
      newpassword: '',
      confirmpassword: ''
  }

  }
  onChangeUserName(e) {
    this.setState({ newusername: e.target.value })

}

onChangeUserEmail(e) {
    this.setState({ newemail: e.target.value })
}

onChangeUserPassword(e) {
    this.setState({ newpassword: e.target.value })
}

onChangeConfirmPassword(e) {
  this.setState({ confirmpassword: e.target.value })
}

  onSubmitUsername(e) {
      e.preventDefault()
      this.state.username = this.state.newusername
  }

  onSubmitEmail(e) {
    e.preventDefault()
    this.state.email = this.state.newemail
  }

  onSubmitPassword(e) {
    e.preventDefault()


  }

    
  render() {
    return (
      <Modal
        style={{marginTop: 100, top: "auto",bottom: "auto",left: "auto",right: "auto",position: "relative",height: 400}} 
        trigger={<Button size='small' style={{marginTop: 5,backgroundColor: "white",color: "black",}}>
        <i className="settings icon" style={{marginRight: 15}}></i>
        Settings
      </Button>
        } centered={false} closeIcon>
        <Modal.Header>Settings</Modal.Header>
        <Modal.Content >
          <Modal.Description>
           <span style={{margin: 50}}>
                Username: {this.state.username}
                <br></br>
                <Modal
                  style={{marginTop: 100, top: "auto",bottom: "auto",left: "auto",right: "auto",position: "relative",height: 200, width: 400}} 
                  trigger ={<Button size='medium' style={{fontSize: 13, backgroundColor: "#2A9D8F",color: "white", marginLeft: 50, marginTop: 20, marginBottom: 20}}>
                <i className="users icon"></i>
                    Update Username
                </Button>} >
                <Modal.Header>Update Username</Modal.Header>
        <Modal.Content >
          <Modal.Description>
          <div>
              <Form size="small" style={{width: 200}} onSubmit={this.onSubmitUsername} >
                <Form.Field
                  label='New Username'
                  control='input'
                  placeholder='New username'
                  value={this.state.newusername} 
                  onChange={this.onChangeUserName}
                />
                 <button className="ui button" style={{backgroundColor: "#2A9D8F", color: "white"}}>Submit</button>
              </Form>
            </div>
           </Modal.Description>
          </Modal.Content>
          </Modal>
            </span>
            <br></br>
            <span style={{margin: 50}}>
                Email: {this.state.email}
                <br></br>
                <Modal 
                style={{marginTop: 100, top: "auto",bottom: "auto",left: "auto",right: "auto",position: "relative",height: 200, width:400}} 
                trigger={<Button size='medium' style={{fontSize: 13, backgroundColor: "#2A9D8F",color: "white", marginLeft: 50, marginTop: 20, marginBottom: 20}}>
                <i className="envelope icon"></i>
                    Update Email Address
                </Button>} >
                <Modal.Header>Update Email Address</Modal.Header>
        <Modal.Content >
          <Modal.Description>
          <div>
              <Form size="small" style={{width: 200}} onSubmit={this.onSubmitEmail} >
                <Form.Field
                  label='New Email Address'
                  control='input'
                  type='email'
                  placeholder='New Email Address'
                  value={this.state.newemail} 
                  onChange={this.onChangeUserEmail}
                />
                 <button className="ui button" style={{backgroundColor: "#2A9D8F", color: "white"}}>Submit</button>
              </Form>
            </div>
           </Modal.Description>
          </Modal.Content>
          </Modal>
            </span>
            <br></br>
            <span style={{margin: 50}}>
                Password:
                <br></br>
                <Modal 
                style={{marginTop: 100, top: "auto",bottom: "auto",left: "auto",right: "auto",position: "relative",height: 300, width:400}} 
                trigger={<Button size='medium' style={{fontSize: 13, backgroundColor: "#2A9D8F",color: "white", marginLeft:50, marginTop: 20, marginBottom: 20}}>
                <i className="lock icon"></i>
                    Update Password
                </Button>} >
                <Modal.Header>Update Password</Modal.Header>
                <Modal.Content >
                <Modal.Description>
                <div>
                <Form size="small" style={{width: 200}} onSubmit={this.onSubmitPassword}>
                <Form.Field
                  label='New Password'
                  control='input'
                  type= 'password'
                  placeholder='New password'  
                  value={this.state.newpassword} 
                  onChange={this.onChangeUserPassword}
                />
                <Form.Field
                  label='Confirm Password'
                  control='input'
                  type= 'password'
                  placeholder='Confirm Password'
                  value={this.state.confirmpassword} 
                  onChange={this.onChangeConfirmPassword}
                />
                 <input type="submit" className="ui button" style={{backgroundColor: "#2A9D8F", color: "white"}} value="Submit"/>
              </Form>
            </div>
           </Modal.Description>
          </Modal.Content>
          </Modal>
            </span>
          </Modal.Description>
        </Modal.Content>
      </Modal>
    );
  }
}

