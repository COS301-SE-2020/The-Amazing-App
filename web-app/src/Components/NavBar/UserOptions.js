import React from "react";
import { Dropdown, Icon, Modal } from "semantic-ui-react";
import image2 from "../../Assets/no_picture.jpg";
import UpdatePreferences from "./UpdatePreferences";
import axios from "axios";
import Cookies from "js-cookie";

const instance = axios
  .get("http://localhost:8000/api/auth/me", {
    headers: { Authorization: "Bearer " + Cookies.get("token") },
  })
  .then((res) => {
    if (res.status == 200) {
      Cookies.set("username", res.data.data.username, { expires: 2 });
    }
  })
  .catch((error) => {
    console.log(error.message);
  });

const trigger = (
  <div>
    <img
      src={image2}
      className="ui avatar image"
      alt="userImage"
      style={{ height: 20, width: 20 }}
    />
    <span style={{ fontSize: 14, color: "white" }}>
      {Cookies.get("username")}
    </span>
  </div>
);

const options = [
  <UpdatePreferences />,
  { key: "sign-out", text: "Sign Out", icon: "sign out" },
];

const UserOptions = () => (
  <Dropdown
    trigger={trigger}
    options={options}
    pointing="top left"
    icon={null}
  />
);

export default UserOptions;
