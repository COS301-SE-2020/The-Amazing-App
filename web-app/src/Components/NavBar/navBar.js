import React from "react";
import { Nav, Navbar } from "react-bootstrap";
import logo from "../../assets/logo.png";
import UserOptions from "./UserOptions";

class NavBar extends React.Component {
  render() {
    const myStyle = {
      fontSize: 16,
      color: "white",
      fontFamily: "'Helvetica Neue', Arial, Helvetica, sans-serif",
    };

    return (
      <Navbar sticky="top" style={{ backgroundColor: "#2A9D8F" }}>
        <Navbar.Brand style={{ marginLeft: 75, color: "white", fontSize: 15 }}>
          <img
            src={logo}
            className="ui small circular image"
            alt="userImage"
            style={{ height: 50, width: 50 }}
          />
        </Navbar.Brand>
        <Nav className="mr-auto"></Nav>
        <Nav
          style={{ marginRight: 100, color: "white", fontWeight: "normal" }}
          className="lead"
          pull
          Right
        >
          <Nav.Link href="/" className="active" style={myStyle} exact>
            Home
          </Nav.Link>
          <Nav.Link className="active" href="dashboard" style={myStyle} exact>
            Dashboard
          </Nav.Link>
          <Nav.Link href="#">
            <UserOptions />
          </Nav.Link>
        </Nav>
      </Navbar>
    );
  }
}

export default NavBar;
