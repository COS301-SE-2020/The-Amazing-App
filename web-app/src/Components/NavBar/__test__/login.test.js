const { Item } = require("semantic-ui-react")
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme,{mount ,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NavBar from '../navBar';
import Footer from '../Footer'
import { BrowserRouter as Router } from "react-router-dom";
import SignIn from '../../Login/SignIn'
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from 'redux';

Enzyme.configure({ adapter: new Adapter() });
const store = createStore(() => [], {}, applyMiddleware());

describe('Login component',()=>
{
    

    it('should not render any buttons',()=>{
        const button = shallow(<Provider  store={store}><Footer />
            </Provider>);
        expect(button.find('Button').length).toBe(0);
        //expect(button_.length).toBe(1);
    });

    it('should render required Divs', () => {
        const calculator = shallow(<Provider  store={store}><Footer />
            </Provider>);  
        // expect(links.length).toBe(0);
        const length = 0;
       // expect(form.find('input').length).toBe(2);
        expect(calculator.find('Div').length).toBe(0);
       
        //expect(form.find('button').length).toBe(1);
       // expect(form.find('p.result').length).toBe(1);
      });

    it('Testing footer text',()=>{
        const footer = shallow(<Provider  store={store}><Footer />
            </Provider>);
        const footer_ = footer.find('Nav.Link Nav.Link');
        const footer__ = true;
        expect(footer__).toBe(true);
    });

   
    


})