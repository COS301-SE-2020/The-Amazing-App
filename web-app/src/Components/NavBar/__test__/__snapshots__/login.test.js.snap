// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component it should render 1`] = `
<nav
  className="navbar navbar-expand navbar-light sticky-top"
  style={
    Object {
      "backgroundColor": "#2A9D8F",
    }
  }
>
  <span
    className="navbar-brand"
    style={
      Object {
        "color": "white",
        "fontSize": 15,
        "marginLeft": 75,
      }
    }
  >
    <img
      alt="userImage"
      className="ui small circular image"
      src="logo.png"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </span>
  <div
    className="mr-auto navbar-nav"
    onKeyDown={[Function]}
  />
  <div
    Right={true}
    className="lead navbar-nav"
    onKeyDown={[Function]}
    pull={true}
    style={
      Object {
        "color": "white",
        "fontWeight": "normal",
        "marginRight": 100,
      }
    }
  >
    <a
      className="nav-link"
      data-rb-event-key="/"
      exact={true}
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "fontFamily": "'Helvetica Neue', Arial, Helvetica, sans-serif",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      <button
        className="ui medium button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "Transparent",
          }
        }
      >
        Home
      </button>
    </a>
    <a
      className="nav-link"
      data-rb-event-key="dashboard"
      exact={true}
      href="dashboard"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "fontFamily": "'Helvetica Neue', Arial, Helvetica, sans-serif",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      <button
        className="ui medium button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "Transparent",
            "border": "2px solid white",
            "color": "white",
          }
        }
      >
        Dashboard
      </button>
    </a>
    <a
      className="nav-link"
      data-rb-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "fontFamily": "'Helvetica Neue', Arial, Helvetica, sans-serif",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      <button
        className="ui medium button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "Transparent",
          }
        }
      >
        Logout
        <i
          className="sign-out"
        />
      </button>
    </a>
    <a
      className="nav-link"
      data-rb-event-key="#"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <button
        className="ui medium button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "Transparent",
          }
        }
      >
        <div
          aria-expanded={false}
          className="ui top left pointing dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div>
            <img
              alt="userImage"
              className="ui avatar image"
              src="no_picture.jpg"
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
            <span
              style={
                Object {
                  "color": "white",
                  "fontSize": 14,
                }
              }
            >
              Tapiwa
            </span>
          </div>
          <div
            className="menu transition"
          >
            <button
              className="ui small button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "color": "black",
                  "marginTop": 5,
                }
              }
            >
              <i
                className="settings icon"
                style={
                  Object {
                    "marginRight": 15,
                  }
                }
              />
              Settings
            </button>
          </div>
        </div>
      </button>
    </a>
  </div>
</nav>
`;
